tests/PHPStan/Analyser/NodeScopeResolverTest.php

```diff
@@ -101,138 +101,148 @@ private static function findTestFiles(): iterable
 
 		if (PHP_VERSION_ID >= 80200) {
-			yield __DIR__ . '/../Rules/Constants/data/bug-8957.php';
+			yield [__DIR__ . '/../Rules/Constants/data/bug-8957.php', []];
 		}
 
 		if (PHP_VERSION_ID >= 80100) {
-			yield __DIR__ . '/../Rules/Comparison/data/bug-9499.php';
+			yield [__DIR__ . '/../Rules/Comparison/data/bug-9499.php', []];
 		}
 
-		yield __DIR__ . '/../Rules/PhpDoc/data/bug-8609-function.php';
-		yield __DIR__ . '/../Rules/Comparison/data/bug-5365.php';
-		yield __DIR__ . '/../Rules/Comparison/data/bug-6551.php';
-		yield __DIR__ . '/../Rules/Variables/data/bug-9403.php';
-		yield __DIR__ . '/../Rules/Methods/data/bug-9542.php';
-		yield __DIR__ . '/../Rules/Functions/data/bug-9803.php';
-		yield __DIR__ . '/../Rules/PhpDoc/data/bug-10594.php';
-		yield __DIR__ . '/../Rules/Classes/data/bug-11591.php';
-		yield __DIR__ . '/../Rules/Classes/data/bug-11591-method-tag.php';
-		yield __DIR__ . '/../Rules/Classes/data/bug-11591-property-tag.php';
-		yield __DIR__ . '/../Rules/Classes/data/mixin-trait-use.php';
+		yield [__DIR__ . '/../Rules/PhpDoc/data/bug-8609-function.php', []];
+		yield [__DIR__ . '/../Rules/Comparison/data/bug-5365.php', []];
+		yield [__DIR__ . '/../Rules/Comparison/data/bug-6551.php', []];
+		yield [__DIR__ . '/../Rules/Variables/data/bug-9403.php', []];
+		yield [__DIR__ . '/../Rules/Methods/data/bug-9542.php', []];
+		yield [__DIR__ . '/../Rules/Functions/data/bug-9803.php', []];
+		yield [__DIR__ . '/../Rules/PhpDoc/data/bug-10594.php', []];
+		yield [__DIR__ . '/../Rules/Classes/data/bug-11591.php', []];
+		yield [__DIR__ . '/../Rules/Classes/data/bug-11591-method-tag.php', []];
+		yield [__DIR__ . '/../Rules/Classes/data/bug-11591-property-tag.php', []];
+		yield [__DIR__ . '/../Rules/Classes/data/mixin-trait-use.php', []];
+
+		yield [
+			__DIR__ . '/data/assert-variable-certainty-on-array.php',
+			[
+				"Certainty of offset 'firstName' on line 29:\nExpected: No\nActual:   Yes\n",
+				"Certainty of variable \$context on line 30:\nExpected: No\nActual:   Yes\n",
+			],
+		];
 	}
 
 	/**
-	 * @return iterable<string, array{string}>
+	 * @return iterable<string, array{string, list<string>}>
 	 */
 	public static function dataFile(): iterable
 	{
 		$base = dirname(__DIR__, 3) . '/';
 		$baseLength = strlen($base);
 
 		$fileHelper = new FileHelper($base);
-		foreach (self::findTestFiles() as $file) {
+		foreach (self::findTestFiles() as $testData) {
+			[$file, $expectedFailures] = $testData;
 			$file = $fileHelper->normalizePath($file);
 
 			$testName = $file;
 			if (str_starts_with($file, $base)) {
 				$testName = substr($file, $baseLength);
 			}
 
-			yield $testName => [$file];
+			yield $testName => [$file, $expectedFailures];
 		}
 	}
 
 	/**
+	 * @param list<string> $expectedFailures
 	 * @dataProvider dataFile
 	 */
-	public function testFile(string $file): void
+	public function testFile(string $file, array $expectedFailures = []): void
```

Comment:
Don't change this signature. Don't change the many `yield` lines. It doesn't make sense to expect failures outside of testing the actual implementation.

There's already TypeInferenceTestCaseTest. Test it there.

Also, add

Reply:
It seems the last part of your comment was not stored...

Reply:
I meant to delete that :)

Reply:
@ondrejmirtes Done, I hope this is what you had in mind.